<a href="https://colab.research.google.com/github/WSmithDR/espol-bootcamp-data_science/blob/m2-project/Plantilla_Proyecto.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Exploración y Segmentación de Clientes
#### Super Bank


1.   Integrante 1:
2.   Integrante 2:



In [66]:
import pandas as pd
sb = pd.read_csv("superbank.csv")

Dimensiones del dataset:

In [67]:
sb.shape

(8950, 18)

Informacion estadisitica básica:

In [68]:
sb.describe()

BALANCE  BALANCE_FREQUENCY     PURCHASES  ONEOFF_PURCHASES  \
count   8055.000000        8055.000000   8055.000000       8055.000000   
mean    1564.721016           0.877907    992.203722        593.487287   
std     2077.820226           0.235817   2077.103541       1665.083373   
min        0.000000           0.000000      0.000000          0.000000   
25%      128.770565           0.888889     35.955000          0.000000   
50%      876.577775           1.000000    356.150000         40.700000   
75%     2063.888893           1.000000   1106.160000        575.250000   
max    19043.138560           1.000000  41050.400000      40761.250000   

       INSTALLMENTS_PURCHASES  CASH_ADVANCE  PURCHASES_FREQUENCY  \
count             8055.000000   8055.000000          8055.000000   
mean               411.606940    987.140035             0.489973   
std                915.321611   2126.252235             0.401603   
min                  0.000000      0.000000             0.000000   
25%                  0.000000      0.000000             0.083333   
50%                 87.500000      0.000000             0.500000   
75%                468.050000   1116.400384             0.916667   
max              22500.000000  47137.211760             1.000000   

       ONEOFF_PURCHASES_FREQUENCY  PURCHASES_INSTALLMENTS_FREQUENCY  \
count                 8055.000000                       8055.000000   
mean                     0.201413                          0.366736   
std                      0.297820                          0.397965   
min                      0.000000                          0.000000   
25%                      0.000000                          0.000000   
50%                      0.083333                          0.166667   
75%                      0.272727                          0.750000   
max                      1.000000                          1.000000   

       CASH_ADVANCE_FREQUENCY  CASH_ADVANCE_TRX  PURCHASES_TRX  CREDIT_LIMIT  \
count             8055.000000       8055.000000    8055.000000   8054.000000   
mean                 0.135315          3.202110      14.736934   4469.021711   
std                  0.200593          6.575732      24.815914   3615.216282   
min                  0.000000          0.000000       0.000000     50.000000   
25%                  0.000000          0.000000       1.000000   1600.000000   
50%                  0.000000          0.000000       7.000000   3000.000000   
75%                  0.222222          4.000000      17.000000   6100.000000   
max                  1.500000        123.000000     347.000000  30000.000000   

           PAYMENTS  MINIMUM_PAYMENTS  PRC_FULL_PAYMENT       TENURE  
count   8055.000000       7782.000000       8055.000000  8055.000000  
mean    1729.655798        855.152072          0.155901    11.519181  
std     2870.333410       2304.057128          0.294644     1.331682  
min        0.000000          0.019163          0.000000     6.000000  
25%      381.405386        169.007295          0.000000    12.000000  
50%      858.811070        309.367321          0.000000    12.000000  
75%     1898.019790        817.732461          0.142857    12.000000  
max    50721.483360      76406.207520          1.000000    12.000000

In [69]:
sb.head(20)

CUST_ID      BALANCE  BALANCE_FREQUENCY  PURCHASES  ONEOFF_PURCHASES  \
0   C10001          NaN           0.818182      95.40              0.00   
1   C10002  3202.467416                NaN       0.00               NaN   
2   C10003  2495.148862           1.000000        NaN            773.17   
3   C10004          NaN           0.636364    1499.00           1499.00   
4   C10005   817.714335           1.000000      16.00               NaN   
5   C10006          NaN           1.000000    1333.28               NaN   
6   C10007   627.260806           1.000000        NaN           6402.63   
7   C10008  1823.652743           1.000000     436.20              0.00   
8   C10009  1014.926473           1.000000     861.49               NaN   
9   C10010   152.225975           0.545455    1281.60               NaN   
10  C10011  1293.124939                NaN     920.12              0.00   
11  C10012   630.794744           0.818182    1492.18           1492.18   
12  C10013  1516.928620           1.000000    3217.99           2500.23   
13  C10014   921.693369                NaN    2137.93               NaN   
14  C10015          NaN           1.000000       0.00              0.00   
15  C10016  6886.213231           1.000000    1611.70              0.00   
16  C10017  2072.074354           0.875000       0.00              0.00   
17  C10018    41.089489           0.454545     519.00              0.00   
18  C10019  1989.072228           1.000000     504.35            166.00   
19  C10020  3577.970933           1.000000     398.64              0.00   

    INSTALLMENTS_PURCHASES  CASH_ADVANCE  PURCHASES_FREQUENCY  \
0                    95.40      0.000000             0.166667   
1                     0.00   6442.945483             0.000000   
2                     0.00           NaN             1.000000   
3                     0.00    205.788017             0.083333   
4                     0.00      0.000000             0.083333   
5                  1333.28      0.000000             0.666667   
6                   688.38      0.000000             1.000000   
7                   436.20      0.000000                  NaN   
8                   200.00      0.000000             0.333333   
9                     0.00      0.000000             0.166667   
10                  920.12      0.000000             1.000000   
11                    0.00      0.000000             0.250000   
12                  717.76      0.000000             1.000000   
13                 1717.97      0.000000             0.750000   
14                    0.00    346.811390             0.000000   
15                 1611.70   2301.491267             0.500000   
16                    0.00   2784.274703             0.000000   
17                  519.00      0.000000                  NaN   
18                     NaN           NaN             0.666667   
19                  398.64      0.000000             1.000000   

    ONEOFF_PURCHASES_FREQUENCY  PURCHASES_INSTALLMENTS_FREQUENCY  \
0                     0.000000                          0.083333   
1                     0.000000                          0.000000   
2                     1.000000                          0.000000   
3                     0.083333                          0.000000   
4                     0.083333                          0.000000   
5                     0.000000                          0.583333   
6                     1.000000                          1.000000   
7                     0.000000                          1.000000   
8                     0.083333                          0.250000   
9                     0.166667                          0.000000   
10                    0.000000                          1.000000   
11                    0.250000                               NaN   
12                    0.250000                          0.916667   
13                    0.166667                          0.750000   
14                         NaN           

Valores faltantes por columna:

In [70]:
sb.isnull().sum()

CUST_ID                                0
BALANCE                              895
BALANCE_FREQUENCY                    895
PURCHASES                            895
ONEOFF_PURCHASES                     895
INSTALLMENTS_PURCHASES               895
CASH_ADVANCE                         895
PURCHASES_FREQUENCY                  895
ONEOFF_PURCHASES_FREQUENCY           895
PURCHASES_INSTALLMENTS_FREQUENCY     895
CASH_ADVANCE_FREQUENCY               895
CASH_ADVANCE_TRX                     895
PURCHASES_TRX                        895
CREDIT_LIMIT                         896
PAYMENTS                             895
MINIMUM_PAYMENTS                    1168
PRC_FULL_PAYMENT                     895
TENURE                               895
dtype: int64

Se observa 895 como una cantidad comun de valores nulos que poseen las mayorias de columnas, lo que nos lleva a preguntarnos

## 1. Análisis Exploratorio de Datos




## 1.1 Tratamiento de Datos Faltantes

In [71]:
sb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8950 entries, 0 to 8949
Data columns (total 18 columns):
 #   Column                            Non-Null Count  Dtype  
---  ------                            --------------  -----  
 0   CUST_ID                           8950 non-null   object 
 1   BALANCE                           8055 non-null   float64
 2   BALANCE_FREQUENCY                 8055 non-null   float64
 3   PURCHASES                         8055 non-null   float64
 4   ONEOFF_PURCHASES                  8055 non-null   float64
 5   INSTALLMENTS_PURCHASES            8055 non-null   float64
 6   CASH_ADVANCE                      8055 non-null   float64
 7   PURCHASES_FREQUENCY               8055 non-null   float64
 8   ONEOFF_PURCHASES_FREQUENCY        8055 non-null   float64
 9   PURCHASES_INSTALLMENTS_FREQUENCY  8055 non-null   float64
 10  CASH_ADVANCE_FREQUENCY            8055 non-null   float64
 11  CASH_ADVANCE_TRX                  8055 non-null   float64
 12  PURCHA

In [72]:
sb.isnull().sum()

CUST_ID                                0
BALANCE                              895
BALANCE_FREQUENCY                    895
PURCHASES                            895
ONEOFF_PURCHASES                     895
INSTALLMENTS_PURCHASES               895
CASH_ADVANCE                         895
PURCHASES_FREQUENCY                  895
ONEOFF_PURCHASES_FREQUENCY           895
PURCHASES_INSTALLMENTS_FREQUENCY     895
CASH_ADVANCE_FREQUENCY               895
CASH_ADVANCE_TRX                     895
PURCHASES_TRX                        895
CREDIT_LIMIT                         896
PAYMENTS                             895
MINIMUM_PAYMENTS                    1168
PRC_FULL_PAYMENT                     895
TENURE                               895
dtype: int64

In [73]:
sb.head(10)

CUST_ID      BALANCE  BALANCE_FREQUENCY  PURCHASES  ONEOFF_PURCHASES  \
0  C10001          NaN           0.818182      95.40              0.00   
1  C10002  3202.467416                NaN       0.00               NaN   
2  C10003  2495.148862           1.000000        NaN            773.17   
3  C10004          NaN           0.636364    1499.00           1499.00   
4  C10005   817.714335           1.000000      16.00               NaN   
5  C10006          NaN           1.000000    1333.28               NaN   
6  C10007   627.260806           1.000000        NaN           6402.63   
7  C10008  1823.652743           1.000000     436.20              0.00   
8  C10009  1014.926473           1.000000     861.49               NaN   
9  C10010   152.225975           0.545455    1281.60               NaN   

   INSTALLMENTS_PURCHASES  CASH_ADVANCE  PURCHASES_FREQUENCY  \
0                   95.40      0.000000             0.166667   
1                    0.00   6442.945483             0.000000   
2                    0.00           NaN             1.000000   
3                    0.00    205.788017             0.083333   
4                    0.00      0.000000             0.083333   
5                 1333.28      0.000000             0.666667   
6                  688.38      0.000000             1.000000   
7                  436.20      0.000000                  NaN   
8                  200.00      0.000000             0.333333   
9                    0.00      0.000000             0.166667   

   ONEOFF_PURCHASES_FREQUENCY  PURCHASES_INSTALLMENTS_FREQUENCY  \
0                    0.000000                          0.083333   
1                    0.000000                          0.000000   
2                    1.000000                          0.000000   
3                    0.083333                          0.000000   
4                    0.083333                          0.000000   
5                    0.000000                          0.583333   
6                    1.000000                          1.000000   
7                    0.000000                          1.000000   
8                    0.083333                          0.250000   
9                    0.166667                          0.000000   

   CASH_ADVANCE_FREQUENCY  CASH_ADVANCE_TRX  PURCHASES_TRX  CREDIT_LIMIT  \
0                0.000000               0.0            2.0           NaN   
1                0.250000               4.0            0.0        7000.0   
2                0.000000               0.0           12.0        7500.0   
3                0.083333               1.0            1.0        7500.0   
4                0.000000               0.0            1.0        1200.0   
5                0.000000               0.0            8.0        1800.0   
6                0.000000               0.0            NaN       13500.0   
7                0.000000               NaN           12.0        2300.0   
8                0.000000               0.0            5.0        7000.0   
9                0.000000               0.0            3.0       11000.0   

      PAYMENTS  MINIMUM_PAYMENTS  PRC_FULL_PAYMENT  TENURE  
0   201.802084               NaN               0.0    12.0  
1  4103.032597       1072.340217               NaN    12.0  
2   622.066742        627.284787               0.0    12.0  
3     0.000000               NaN               0.0    12.0  
4   678.334763        244.791237               0.0    12.0  
5  1400.057770       2407.246035               0.0    12.0  
6  6354.314328        198.065894               NaN    12.0  
7          NaN        532.033990               0.0     NaN  
8   688.278568        311.963409               0.0    12.0  
9  1164.770591        100.302262               0.0     NaN

## 1.2  Detección y Eliminación de Valores atípicos

## 1.3 Transformación de datos

## 1.4 Anallisis Univariado

## 1.5 Generación de Insights

# 2. Clustering

## 2.1 Aplicar Clustering

## 2.2 Caracterización de Clusters